detox/src/devices/drivers/android/genycloud/services/GenyInstanceNaming.js

```diff
@@ -0,0 +1,17 @@
+class GenyInstanceNaming {
+  constructor(nowProvider = () => new Date().getTime()) {
+    this.uniqueSessionId = Number(process.env.DETOX_START_TIMESTAMP);
+    this.nowProvider = nowProvider;
+  }
+
+  generateName() {
+    const uniqueDeviceId = this.nowProvider() - this.uniqueSessionId;
```

Comment:
Looks peculiar. @d4vidi, why do you rely so much on timestamps here and on their difference (arithmetical subtraction) for generating names? Looks like a piece of a puzzle I obviously miss here.

Reply:
Uniqueness is the name of the game! We need to wrap our heads around the fact that cloud instances are visible to any CI agent sharing the same Genymotion account... I figured that using a timestamp down to the millisecond would do the trick in isolating a shared account's instances based on the instance names.

If you nevertheless feel that the unique deviceId (so-called) is nontrivial - I could switch it with an index running from 1 to inf. Would you consider that friendlier? The subtraction was a means to achieve the same end-result with very little effort.

Reply:
Why can't we just create a class for providing Session and Worker IDs and use it?

So far, this trick with `this.nowProvider() - this.uniqueSessionId` gives me bad feelings. Is there any problem with relying on `process.env.JEST_WORKER_ID || '0'` under the hood?

Reply:
I was not aware of the JEST_WORKER_ID option. I suppose it can work, yet it would marry a worker with an instance, which is theoretically suboptimal but should practically be ok.

Reply:
See #2470.